test/helper.lua

```diff
@@ -237,4 +237,23 @@ function helpers.is_expired_debug(_, tuple)
     return true
 end
 
+function helpers.tarantool_version()
+    local major_minor_patch = _G._TARANTOOL:split('-', 1)[1]
+    local major_minor_patch_parts = major_minor_patch:split('.', 2)
+
+    local major = tonumber(major_minor_patch_parts[1])
+    local minor = tonumber(major_minor_patch_parts[2])
+    local patch = tonumber(major_minor_patch_parts[3])
+
+    return major, minor, patch
+end
+
+function helpers.vinyl_is_broken()
+    -- Blocked by https://github.com/tarantool/tarantool/issues/6609
+    local major, minor, patch = helpers.tarantool_version()
+    local broken_v1_10 = major == 1 and minor >= 10 and patch >= 11
+    local broken_v2 = major == 2 and (minor >= 1 or minor <= 7)
+    return broken_v1_10 or broken_v2
```

Comment:
For 3.x vinyl will be broken

Reply:
Versions with fix for #6448:
```
sergeyb@pony:~/sources/MRG/tarantool$ needle='vinyl: fix use of dropped space in deferred DELETE handler'; for branch in 1.9 1.10 2.1 2.2 2.3 2.4 2.5 2.6 2.7 2.8 master; do printf "${branch}\t"; commit_description=$(git log --oneline "origin/${branch}" | grep -F "${needle}"); [ -n "${commit_description}" ] && echo "$(git describe --long --always $(echo "${commit_description}" | cut -d" " -f1)) $(echo "${commit_description}" | cut -d" " -f2-)" || echo "(not found)"; done
1.9     (not found)
1.10    1.10.11-5-g40d8f4df1 vinyl: fix use of dropped space in deferred DELETE handler
2.1     (not found)
2.2     (not found)
2.3     (not found)
2.4     (not found)
2.5     (not found)
2.6     (not found)
2.7     (not found)
2.8     2.8.2-12-g2663a9c6c vinyl: fix use of dropped space in deferred DELETE handler
master  2.10.0-beta1-51-g0428bbcef vinyl: fix use of dropped space in deferred DELETE handler
```

Updated function to detect Tarantool version with broken vinyl:
```lua
--- a/test/helper.lua
+++ b/test/helper.lua
@@ -249,10 +249,15 @@ function helpers.tarantool_version()
 end
 
 function helpers.vinyl_is_broken()
-    -- Blocked by https://github.com/tarantool/tarantool/issues/6609
+    -- Blocked by https://github.com/tarantool/tarantool/issues/6448
     local major, minor, patch = helpers.tarantool_version()
-    local broken_v1_10 = major == 1 and minor >= 10 and patch >= 11
-    local broken_v2 = major == 2 and (minor >= 1 or minor <= 7)
+
+    -- Since Tarantool 1.10.11.
+    local broken_v1_10 = major >= 1 and (minor > 10 or minor == 10 and patch >= 11)
+
+    -- Since Tarantool 2.1.7.
+    local broken_v2 = major >= 2 and (minor > 1 or minor == 1 and patch >= 7)
+
     return broken_v1_10 or broken_v2
 end
 
```

Tests for function `vinyl_is_broken()`:

```lua
local function is_vinyl_broken(major, minor, patch)
    --local major, minor, patch = helpers.tarantool_version()
    --local broken_v1_10 = major == 1 and minor >= 10 and patch >= 11
    --local broken_v2 = major == 2 and (minor >= 1 or minor <= 7)

    -- Since Tarantool 1.10.11.
    local broken_v1_10 = major >= 1 and (minor > 10 or minor == 10 and patch >= 11)

    -- Since Tarantool 2.1.7.
    local broken_v2 = major >= 2 and (minor > 1 or minor == 1 and patch >= 7)
    return broken_v1_10 or broken_v2
end

assert(is_vinyl_broken(1, 10, 11) == true)
assert(is_vinyl_broken(1, 10, 10) == false)
assert(is_vinyl_broken(1, 10, 12) == true)
assert(is_vinyl_broken(1, 9, 3) == false)
assert(is_vinyl_broken(1, 11, 4) == true)

assert(is_vinyl_broken(2, 1, 8) == true)
assert(is_vinyl_broken(2, 1, 7) == true)
assert(is_vinyl_broken(2, 1, 6) == false)
assert(is_vinyl_broken(2, 2, 0) == true)
assert(is_vinyl_broken(2, 0, 7) == false)

assert(is_vinyl_broken(3, 0, 1) == false)
assert(is_vinyl_broken(3, 1, 1) == false)
```

Reply:
```
tarantool> assert(is_vinyl_broken(3, 1, 1) == false)
---
- true
...

tarantool> assert(is_vinyl_broken(3, 2, 1) == false)
---
- error: assertion failed!
...
```

Reply:
I have updated a snippet with asserts and function in test/helpers.lua. Seems you have used an old version, that was incorrect.